MobiusCore/Source/EffectHandlers/EffectHandler.swift

```diff
@@ -41,21 +38,31 @@ final public class EffectHandler<Effect, Event> {
     /// that should be sent as input to the `handle` function. If it cannot handle the effect, return `nil`
     /// - Parameter handleEffect: Handle effects which satisfy `canHandle`.
     /// - Parameter stopHandling: Tear down any resources being used by this effect handler.
-    public init<EffectPayload>(
+    public convenience init<EffectPayload>(
         canHandle: @escaping (Effect) -> EffectPayload?,
         handle: @escaping (EffectPayload, @escaping Consumer<Event>) -> Void,
+        stopHandling disposable: @escaping () -> Void = {}
+    ) {
+        self.init(
+            handleEffect: { effect in
+               if let payload = canHandle(effect) {
+                   return { dispatch in
+                       handle(payload, dispatch)
+                   }
+               } else {
+                   return nil
+               }
+            },
+            stopHandling: disposable
+        )
+    }
+
+    init(
+        handleEffect: @escaping (Effect) -> ((@escaping Consumer<Event>) -> Void)?,
```

Comment:
I feel like `handleEffect` is a confusing name, and it's making the static merging `handleEffect` function harder to understand. AFAIU, this function isn't really handling the effect, but determining if it can be handled by this effect handler, and if so, what code should run in order to handle it. The actual execution of that code happens later. If that's not correct (and side effects can happen during the evaluation of `handleEffect`), then the merging function looks broken. 

Maybe something like `selectHandler` could be a better name? It's not completely clear to me that this class is really an `EffectHandler` either, it may be more of an effect router. Either things are a little muddled here, or there are things I'm not understanding.